deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/Convolution1DLayer.java

```
@@ -0,0 +1,156 @@
+            throw new IllegalStateException("Invalid input for 1D CNN layer (layer index = " + layerIndex +
+                    ", layer name = \"" + getLayerName() + "\"): expect RNN input type with size > 0. Got: " + inputType);
+        }
+
+        return InputType.recurrent(nOut);
+    }
+
+    @Override
+    public void setNIn(InputType inputType, boolean override){
+        if (inputType == null || inputType.getType() != InputType.Type.RNN) {
+            throw new IllegalStateException("Invalid input for 1D CNN layer (layer name = \"" + getLayerName() + "\"): expect RNN input type with size > 0. Got: " + inputType);
+        }
+
+        if (nIn <= 0 || override) {
+            InputType.InputTypeRecurrent r = (InputType.InputTypeRecurrent) inputType;
+            this.nIn = r.getSize();
+        }
+    }
+
+    @Override
+    public InputPreProcessor getPreProcessorForInputType(InputType inputType) {
+        if(inputType == null ){
+            throw new IllegalStateException("Invalid input for Convolution1D layer (layer name=\"" + getLayerName() + "\"): input is null");
+        }
+
+        return InputTypeUtil.getPreprocessorForInputTypeRnnLayers(inputType, getLayerName());
+    }
+
+    public static class Builder extends ConvolutionLayer.Builder {
+        public Builder(int kernelSize, int stride, int padding) {
+            this();
+            this.kernelSize[0] = kernelSize;
+            this.stride[0] = stride;
+            this.padding[0] = padding;
+        }
+
+        public Builder(int kernelSize, int stride) {
+            this();
+            this.kernelSize[0] = kernelSize;
+            this.stride[0] = stride;
+        }
+
+        public Builder(int kernelSize) {
+            this();
+            this.kernelSize[0] = kernelSize;
+        }
+
+        public Builder() {
+            this.kernelSize = new int[] { 5, 1 };
+            this.stride = new int[] { 1, 1 };
```

Where does the 5 come from? I feel like we should just throw an exception here.